src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -120,14 +136,6 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     setMessage(undefined);
     try {
       const newTransaction = await fn(payload);
-      // for multiple error messages
-      newTransaction.messages?.forEach((newMessage) => {
-        const { class: type, message: msg } = newMessage;
-        if (type === MessageType.ERROR) {
-          // error event
-          events?.afterError?.(getEventContext(newTransaction), { message: msg });
-        }
-      });
```

Comment:
This is not how v2 expects the afterError hook to be called. Moved and refactored to match parity.